aten/src/ATen/native/cuda/Distributions.cu

```diff
@@ -69,7 +69,14 @@ struct curand_uniform_wrapper {
   curandStatePhilox4_32_10_t &state;
   __device__ curand_uniform_wrapper(curandStatePhilox4_32_10_t &state): state(state) {}
   __device__ float operator()() {
-    return curand_uniform(&state);
+    auto val = curand_uniform(&state);
```

Comment:
This won't work, because `val` is float, and you are static_casting and not reinterpret-casting it to uint64_t, which means that it'll either become 0 or 1. 
```suggestion
    auto val = curand_uniform(&state);
```
    uint val = curand(&state); //need just bits
    constexpr auto MASK = static_cast<uint>((static_cast<uint64_t>(1) << std::numeric_limits<float>::digits) - 1);//MASK should be uint, int64 arithmetic is much slower. In honesty, even static_cast<uint64_t>(1) is not needed because digist for float is 24, so 32 bits is enough, but it does not matter 
    constexpr auto DIVISOR = static_cast<float>(1) / (static_cast<uint>(1) << std::numeric_limits<float>::digits);//uint64_t is not needed here either
  return (val & MASK) * DIVISOR;
```

Reply:
thanks. I assumed tests would catch any gross errors :(

Reply:
adopted in 7257949. Also rebased off master to hopefully clear the CI errors

Reply:
IT's actually concerning that the tests don't catch those errors.

Reply:
It's uint32_t probably to fix windows builds, sorry.